File: DrawPT.Common/Models/Game/Room.cs
namespace DrawPT.Common.Models
{
    public class Room
    {
        public Guid Id { get; set; }
        public string Code { get; set; }

        public string Name { get; set; }

        public int PlayerLimit { get; set; }

        public List<Player> Players { get; set; }

        public bool IsGameStarted { get; set; }
        public bool IsPrivate { get; set; }

        public int CurrentRound { get; set; }
        public int TotalRounds {  get; set; }

        public Room()
        {
            Id = Guid.NewGuid();
            PlayerLimit = 6;
            Players = new List<Player>();
            TotalRounds = 5;
            RegenerateCode();
        }

        public void RegenerateCode()
        {
            var random = new Random();
            var joinCode = new string(Enumerable.Repeat("ABCDEFGHIJKLMNOPQRSTUVWXYZ", 4)
                .Select(s => s[random.Next(s.Length)]).ToArray());
            Code = joinCode;
        }
    }
}

File: DrawPT.Vue/src/stores/room.ts
import { computed, ref } from 'vue'
import { defineStore } from 'pinia'
import type { Room } from '@/models/room'
import type { Player } from '@/models/player'
import type { GameState } from '@/models/gameModels'

export const useRoomStore = defineStore('room', () => {
  const room = ref({
    id: '',
    code: '',
    name: '',
    playerLimit: 8,
    players: [],
    isGameStarted: false,
    isPrivate: false,
    currentRound: 0,
    totalRounds: 7
  } as Room)

  const players = computed((): Player[] => room.value.players || [])
  const successfullyJoined = ref(false)

  function setSuccessfullyJoined(success: boolean) {
    successfullyJoined.value = success
  }

  function updateRoomCode(code: string) {
    room.value.code = code
  }

  function addPlayer(player: Player) {
    room.value.players.push(player)
  }

  function removePlayer(player: Player) {
    const index = room.value.players.findIndex((p) => p.id === player.id)
    if (index !== -1) {
      room.value.players.splice(index, 1)
    }
  }

  function startGame() {
    room.value.isGameStarted = true
  }

  function clearRoom() {
    room.value.id = ''
    room.value.code = ''
    room.value.name = ''
    room.value.playerLimit = 8
    room.value.players = []
    room.value.isGameStarted = false
    room.value.isPrivate = false
    room.value.currentRound = 0
    room.value.totalRounds = 7
    successfullyJoined.value = false
  }

  return {
    setSuccessfullyJoined,
    updateRoomCode,
    clearRoom,
    startGame,
    addPlayer,
    removePlayer,
    room,
    players,
    successfullyJoined
  }
})
